lib/puppet/face/facts.rb

```diff
@@ -35,4 +35,48 @@
 
   deactivate_action(:destroy)
   deactivate_action(:search)
+
+  action(:upload) do
+    summary _("Upload local facts to the puppet master.")
+    description <<-'EOT'
+      Reads facts from the local system using the `facter` terminus, then
+      saves the returned facts using the rest terminus.
+    EOT
+    returns "Nothing."
+    notes <<-'EOT'
+      This action requires that the puppet master's `auth.conf` file
+      allow save access to the `facts` REST terminus. Puppet agent does
+      not use this facility, and it is turned off by default. See
+      <https://puppet.com/docs/puppetserver/latest/config_file_auth.html>
+      for more details.
+    EOT
```

Comment:
@gguillotte should we be referring to `puppet server` instead or some variation?

Reply:
I can't really tell from the context. Is this feature available only on Puppet Server masters, and/or are we testing or supporting this with Puppet Server only? If so, I'd specify that, like on L47:

```
      This action requires that Puppet Server's `auth.conf` file 
```

Reply:
The REST endpoint would be available in both, but we're only changing the default auth.conf in puppetserver, right @Sharpie?

Reply:
The default `auth.conf` for Puppet Server is updated over in puppetlabs/puppetserver#1578. Given that Rack and the ruby-based auth.conf are both deprecated, I think it makes more sense to update these docs to say "Puppet Server" and just leave Rack masters out of it.